harness/determined/pytorch/_lr_scheduler.py

```diff
@@ -67,8 +65,10 @@ def __init__(
             frequency:
                 Sets the frequency at which the batch and epoch step modes get triggered.
         """
-        check.check_not_none(scheduler)
-        check.check_isinstance(step_mode, LRScheduler.StepMode)
+        if scheduler is None:
+            raise ValueError("scheduler must not be None.")
+        if not isinstance(step_mode, LRScheduler.StepMode):
+            raise TypeError(f"step_mode must be an LRScheduler.StepMode. Got {type(step_mode)}.")
```

Comment:
I would consider it more idiomatic to do `step_mode = LRScheduler.StepMode(step_mode)` which will promote specific values (like `"MANUAL_STEP"` automatically).

But I'm not opposed to leaving this in place because it is a little stricter and nobody has ever complained about it.